beacon-chain/sync/service.go

```diff
@@ -0,0 +1,80 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared"
+)
+
+var _ = shared.Service(&RegularSync{})
+
+// RegularSync service is responsible for handling all run time p2p related operations as the
+// main entry point for network messages.
+type RegularSync struct {
+	ctx context.Context
+	p2p p2p.Composite
+}
+
+// Start the regular sync service by initializing all of the p2p sync handlers.
+func (r *RegularSync) Start() {
+	// Register RPC handlers.
```

Comment:
maybe registering the rpc handlers and pubsub subscribers, can be their own separate methods

Reply:
Good idea